web/src/components/vis/PlsdaPage/PlsdaPage.vue

```diff
@@ -35,85 +36,55 @@ export default {
+    const ready = computed(() => {
+      const pcaReady = store.getters.ready(props.id, 'plsda_scores');
+      const loadingsReady = store.getters.ready(props.id, 'plsda_loadings');
       return pcaReady && loadingsReady;
-    },
-    loadings() {
-      return this.maybeData(['loadings'], []);
-    },
-    pcCoords() {
-      return this.maybeData(['scores', 'x'], []);
-    },
-    eigenvalues() {
-      return this.maybeData(['scores', 'sdev'], []);
-    },
-    rowLabels() {
-      return this.maybeData(['rows'], []);
-    },
-    groupLabels() {
-      return this.maybeData(['labels'], {});
-    },
-    columns() {
-      if (this.dataset?.column?.data) {
-        return this.dataset.column.data;
-      }
-      return [];
-    },
+    });
+    const loadings = computed(() => plot.value.data?.loadings || []);
+    const pcCoords = computed(() => plot.value.data?.scores.x || []);
+    const eigenvalues = computed(() => plot.value.data?.scores.sdev || []);
+    const rowLabels = computed(() => plot.value.data?.rows || []);
+    const groupLabels = computed(() => plot.value.data?.labels || {});
+    const columns = computed(() => dataset.value?.column.data || []);
+    const groupLevels = computed(() => dataset.value?.groupLevels || []);
 
-    groupLevels() {
-      if (this.dataset?.groupLevels) {
-        return this.dataset.groupLevels;
+    watchEffect(() => {
+      const pcX = Number.parseInt(controls.pcXval, 10);
+      if (!Number.isNaN(pcX)) {
+        controls.pcX = pcX;
       }
-      return [];
-    },
-  },
-  watch: {
-    pcXval: {
-      handler(val) {
-        const pcX = Number.parseInt(val, 10);
-        if (!Number.isNaN(pcX)) {
-          this.pcX = pcX;
-        }
-      },
-      immediate: true,
-    },
-    pcYval: {
-      handler(val) {
-        const pcY = Number.parseInt(val, 10);
-        if (!Number.isNaN(pcY)) {
-          this.pcY = pcY;
-        }
-      },
-      immediate: true,
-    },
-    numComponentsVal: {
-      handler(val) {
-        const numComponents = Number.parseInt(val, 10);
-        if (!Number.isNaN(numComponents)) {
-          this.numComponents = numComponents;
-        }
-      },
-      immediate: true,
-    },
-  },
-  methods: {
-    maybeData(keys, dflt) {
-      const {
-        plot,
-      } = this;
-      if (keys.length === 2) {
-        return plot.data ? plot.data[keys[0]][keys[1]] : dflt;
+    });
+    watchEffect(() => {
+      const pcY = Number.parseInt(controls.pcYval, 10);
+      if (!Number.isNaN(pcY)) {
+        controls.pcY = pcY;
       }
-      return plot.data ? plot.data[keys[0]] : dflt;
-    },
+    });
+    watchEffect(() => {
+      const numComponents = Number.parseInt(controls.numComponentsVal, 10);
+      if (!Number.isNaN(numComponents)) {
+        controls.numComponents = numComponents;
+      }
+    });
+
+    return {
+      plot,
+      changePlotArgs,
+      ...toRefs(controls),
```

Comment:
as a matter of style, if a group of reactive properties are coupled such that you think they all belong within the same `const controls = reactive({ ... })`, then it would probably make sense to `return { controls };`

Then you can use  `{{ controls.whatever }}` in the template. 

I consider `toRefs()` to be a tool for composition functions and not really something you'll use in SFCs.  

Again, this is more of a style thing.  I'm not sure if @dchiquit chose `toRefs` intentionally or thought it was required for all `reactive()` objects.

Reply:
You can also sorta take this with a grain of salt.  Spread (`...`) is my new pet peeve, mostly because it's hard to know/understand how babel with transform it.  I've become increasingly comfortable with spread over the last year and I've caught myself using it in nested loops and other dumb places.  I'm sorta hedging towards "we should use fewer babel features". 

Anyway, I'm projecting here is my point.

Reply:
I was mostly trying to minimize footprint and save verbosity, but now that I consider it properly `v-model="controls.showEllipses"` is more legible to me than `v-model="showEllipses"`. I will change that.